### ISSUE TITLE
DefaultThemeRenderContext.urlTo isn't used for Markdown documents

### ISSUE THREAD

**Issue by nex3** [TypeStrong/typedoc#1731]
## Expected Behavior

When I override `DefaultThemeRenderContext.urlTo` in a custom theme, I'd expect that to control the construction of all links in the generated document.

## Actual Behavior

The `urlTo` callback isn't invoked for stand-alone Markdown documents like `README.md`, leaving me with no way to control the URLs generated by those documents other than ugly monkey-patching or post-processing.

## Steps to reproduce the bug

```js
class MyRenderContext extends DefaultThemeRenderContext {
  // Relative URLs don't work well for index pages since they can be rendered
  // at different directory levels, so we just convert all URLs to absolute to
  // be safe.
  urlTo = bind(function(context, reflection) {
    if (!reflection.url) return undefined;

    const relative = context.theme.markedPlugin.getRelativeUrl(reflection.url);
    const absolute = new URL(
      relative,
      `relative:///documentation/js-api/${context.theme.markedPlugin.location}`
    );
    absolute.pathname = absolute.pathname
      .replace(/\.html$/, '')
      .replace(/\/index$/, '');
    return absolute.toString().replace(/^relative:\/\//, '');
  }, this);
}
```

## Environment

-   Typedoc version: 0.22.4
-   TypeScript version: 4.3.5
-   Node.js version: 16.4.2
-   OS: Linux

**Comment by Gerrit0** [comment]
Even if links in the readme were passed through the renderer context, you still wouldn't be able to get it from `urlTo`. `urlTo` only accepts reflections. You would have to override [DefaultThemeRenderContext.relativeURL](https://typedoc.org/api/classes/DefaultThemeRenderContext.html#relativeURL) instead.

I'm not opposed to this, but I think the way this really needs to be done is to replace the ad-hoc link/page creation with a proper router, and let themes provide a custom router.

Changing this to enhancement - This is a new feature, not a defect with the existing implementation.

**Comment by Gerrit0** [comment]
This totally was a bug, I just didn't understand it until today... and it happens to have been fixed by refactoring done for 0.23. `context.urlTo` will be invoked for all reflection links now.
